rfcs/0107-version-normalization.md

```diff
@@ -0,0 +1,223 @@
+builtins.parseDrvName "mpv-0.35.15"
+{ name = "mpv"; version = "0.35.15"; }
+
+# expected: { name = "mpv"; version = "unstable-2021-05-03"; }
+nix-repl> builtins.parseDrvName "mpv-unstable-2021-05-03"
+builtins.parseDrvName "mpv-unstable-2021-05-03"
+{ name = "mpv-unstable"; version = "2021-05-03"; }
+```
+
+It happens because the `version` attribute in the set returned by
+`builtins.parseDrvName` always starts with a digit. It happens by a deliberate
+design decision, and as such it should not be regarded as a "bug"; therefore, we
+should strive to follow it, neither circumventing nor ignoring it.
+
+Further, the `version` attribute should be crafted to satisfy the expected
+upgrading semantics stated in the manual pages, as effectively implemented by
+`builtins.compareVersions` -- even when the raw version of the original program
+does not meet this expectation.
+
+Besides, Nixpkgs should provide a consistent interface to external package
+monitoring services like [Repology](https://repology.org/).
+
+This document describes a format suitable to fix these issues, while keeping it
+understandable and striving for simplicity.
+
+# Detailed design
+[design]: #detailed-design
+
+## Terminology
+
+_Disclaimer_: whereas some of the terms enumerated below are borrowed from
+[pkgsrc guide](https://www.netbsd.org/docs/pkgsrc/) plus a bit of terminology
+employed by git, this document aims to be general, not being overly attached to
+them.
+
+- Program denotes the piece of software to be fetched and processed via Nixpkgs.
+  - This term makes no distinction about the immediate or intended uses of the
+    program; it can range from non-interactive programs to full GUI
+    applications, even including filesets intended as input to other programs
+    (such as firmwares and fonts).
+- Team denotes the maintainers of the program.
+  - This term makes no distinction among the various models of organization of a
+    team; it ranges from a solitary programmer to a business company - and
+    everything inbetween.
+- Source denotes the origin of the program.
+  - This term makes no distinction among precompiled, binary-only or high-level
+    code distributions.
+- Snapshot denotes the source of the program taken at a point of the time.
+  - Labeled snapshot denotes any snapshot explicitly labeled by its team.
+  - Unlabeled snapshot denotes any snapshot not explicitly labeled by its team.
+- Release denotes any distributed snapshot, as defined by its team.
+- Branch denotes a logical sequence of snapshots, as identified by the program's
+  team;
+  - Usually these branches are denoted by names such as `stable`, `master`,
+    `unstable`, `trunk`, `experimental`, `staging`, `X.Y` (where `X` and
+    `Y` are numbers) etc.
+
+## Design
+
+- For a labeled snapshot:
+  - `version` should be constituted of the version of the snapshot, as defined
+    by the program team, without any alphabetical characters (e.g. "v", "rel")
```

Comment:
is `rel` popular? i have never seen it. do you have exapmles?

what about versions like `R12`? can we add that here and in the examples? maybe replace `rel`

examples:

- [VapourSynth](https://www.vapoursynth.com/) R59
- [VapourSynth-Editor](https://github.com/YomikoR/VapourSynth-Editor) [r19-mod-5.3](https://github.com/YomikoR/VapourSynth-Editor/releases/tag/r19-mod-5.3)
- [erlangR21](https://search.nixos.org/packages?channel=22.05&from=0&size=50&sort=relevance&type=packages&query=erlang)
- [LS-DYNA](https://en.wikipedia.org/wiki/LS-DYNA) R10.1
- [MATLAB](https://en.wikipedia.org/wiki/MATLAB) R2022a
- [Oracle E-Business Suite](https://en.wikipedia.org/wiki/Oracle_Applications#Oracle_E-Business_Suite) Release 12 (R12)

Reply:
I believe this is clear enough. The listing is not exaustive, only exemplificative.

But I can add it, no problem.

Reply:
I would add R and remove rel. Then we have the most common as examples.